File: src/_1ProgrammingBasicsWithJavaMay2023/_8Exam/_2OtherExam/_04Exam.java
package _1ProgrammingBasicsWithJavaMay2023._8Exam._2OtherExam;

import java.util.Scanner;

public class _04Exam {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = Integer.parseInt(scanner.nextLine());
        int countTo3 = 0;
        int countTo4 = 0;
        int countTo5 = 0;
        int countTo6 = 0;
        double totalGrade = 0;
        for (int i = 0; i < n; i++) {
            double grade = Double.parseDouble(scanner.nextLine());
            if (grade >= 5.00) {
                countTo6++;
            } else if (grade >= 4.00) {
                countTo5++;
            }  else if (grade >= 3.00) {
                countTo4++;
            } else {
                countTo3++;
            }
            totalGrade += grade;
        }
        System.out.printf("Top students: %.2f%%%n", (1.0 * countTo6 / n) * 100.00);
        System.out.printf("Between 4.00 and 4.99: %.2f%%%n",(1.0 * countTo5 / n) * 100.00);
        System.out.printf("Between 3.00 and 3.99: %.2f%%%n",(1.0 * countTo4 / n) * 100.00);
        System.out.printf("Fail: %.2f%%%n",(1.0 * countTo3 / n) * 100.00);
        System.out.printf("Average: %.2f%n", totalGrade / n);
    }
}

File: src/_1ProgrammingBasicsWithJavaScriptNovember2024/_8Exam/_1MainExam/_04Exam.js
function exam(input) {
    const n = Number(input[0]);
    let countTo3 = 0;
    let countTo4 = 0;
    let countTo5 = 0;
    let countTo6 = 0;
    let totalGrade = 0;

    let index = 1;
    for (let i = 0; i < n; i++) {
        const grade = Number(input[index++]);
        if (grade >= 5.00) {
            countTo6++;
        } else if (grade >= 4.00) {
            countTo5++;
        }  else if (grade >= 3.00) {
            countTo4++;
        } else {
            countTo3++;
        }
        totalGrade += grade;
    }
    const gradeTo6 = (countTo6 / n) * 100.00;
    const gradeTo5 = (countTo5 / n) * 100.00;
    const gradeTo4 = (countTo4 / n) * 100.00;
    const gradeTo3 = (countTo3 / n) * 100.00;
    const average = totalGrade / n;
    console.log(`Top students: ${gradeTo6.toFixed(2)}%`);
    console.log(`Between 4.00 and 4.99: ${gradeTo5.toFixed(2)}%`);
    console.log(`Between 3.00 and 3.99: ${gradeTo4.toFixed(2)}%`);
    console.log(`Fail: ${gradeTo3.toFixed(2)}%`);
    console.log(`Average: ${average.toFixed(2)}`);
}

exam((["10",
    "3.00",
    "2.99",
    "5.68",
    "3.01",
    "4",
    "4",
    "6.00",
    "4.50",
    "2.44",
    "5"])
)

exam((["6",
    "2",
    "3",
    "4",
    "5",
    "6",
    "2.2"])
)